### ISSUE TITLE
Place AR.js (w A-Frame) at the bottom of the webpage

### ISSUE THREAD

**Issue by Angelo8828** [jeromeetienne/AR.js#294]
*First of all, sorry guys because I know you had a hard-time already maintaning and fixing AR.js and maybe I'm about to add one. Thanks for the great work guys!*

**Do you want to request a *feature* or report a *bug*?**
Maybe a **feature request**.

So here's my problem. I am currently utilizing AR.js for a work project. I used [https://github.com/jeromeetienne/AR.js/blob/master/aframe/examples/multiple-independent-markers.html](https://github.com/jeromeetienne/AR.js/blob/master/aframe/examples/multiple-independent-markers.html) as a guide and it works perfectly fine. The problem that I encountered is that when I placed the AR.js/A-Frame scripts after the body tag, there has been an error. **I needed to be able to place the scripts after the body tag** just like a normal JS file for the sake of **page load optimization**.

**What is the current behavior? If the current behavior is a bug, please provide the steps to reproduce.**

![ar js bug](https://user-images.githubusercontent.com/11204084/34555078-efd481e2-f16a-11e7-9ac3-97d50972472c.png)

It seems that the `addEventListener` in the *aframe.min.js* already needs something after the page has been loaded.

**If this is a feature request, what is motivation or use case for changing the behavior?**

Stated already above.

**Please mention other relevant information such as the browser version, Operating System and Device Name**

Chrome 63, Ubuntu 14.04 || Chrome 63, Windows 10

EDIT: Changed my objective

**Comment by Angelo8828** [comment]
Sorry, the problem is on A-Frame, not on AR.js.

**Comment by Angelo8828** [comment]
Reopened the issue because of my wrong diagnosis of the problem. I created [a repository](https://github.com/Angelo8828/sample-arjs-implementation) to clearly demonstrate the problem. The current code for the repository works.

Now, this are the problems ([please check the index.html](https://github.com/Angelo8828/sample-arjs-implementation/blob/master/index.html)).
1. If we will move only the `aframe.min.js` to the bottom of the HTML page after the body tag ([at line 37](https://github.com/Angelo8828/sample-arjs-implementation/blob/master/index.html#L37)), the screenshot in the first post will happen.
2. If we will move only the `aframe-ar.js` to the bottom instead of the `aframe.min.js`. No error will be returned but the marker detection (or sometimes, the camera if it has been cached already) won't work.
3. If we will move both of scripts at the bottom, the same thing in No.1 will happen.

**Comment by Angelo8828** [comment]
UPDATES:

I thought to see and screenshot again the error messages in console to check what JS file generates the error. 

![From imgur](https://i.imgur.com/C8a4pCg.png)

The error comes from [aframe.min.js](https://github.com/jeromeetienne/AR.js/blob/master/aframe/examples/vendor/aframe/build/aframe.min.js). I already diagnosed this earlier so that's why I closed the issue because I thought the problem is on A-Frame, not AR.js. 

After some analysis, I knew that the [aframe-v0.6.1-three-r86.js](https://github.com/jeromeetienne/AR.js/blob/master/aframe/examples/vendor/aframe/build/aframe-v0.6.1-three-r86.js) is the unminified version of `aframe.min.js` (judging from the same file sizes). I directly included the script and the exact error has been displayed. The line which generates the error is at line 66049 (`sceneEl.addEventListener('enter-vr', this.onEnterVR);`).
